validator/validator.go

```diff
@@ -9,32 +9,30 @@ import (
 
 	addr "github.com/filecoin-project/go-address"
 	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/go-state-types/big"
 )
 
 // NewValidatorFromString parses a string containing of validator address and multiaddress separated by "@".
 //
 // Example of validator strings:
 // - t1wpixt5mihkj75lfhrnaa6v56n27epvlgwparujy@/ip4/127.0.0.1/tcp/10000/p2p/12D3KooWJhKBXvytYgPCAaiRtiNLJNSFG5jreKDu2jiVpJetzvVJ
 // FIXME: Consider using json serde for this to support multiple multiaddr for validators.
-func NewValidatorFromString(s string) (Validator, error) {
+func NewValidatorFromString(s string) (*Validator, error) {
```

Comment:
Let's already encode also the weight in the string and parse it out here.

Reply:
Fixed